### PR TITLE
TypeScript Configuration

### PR DESCRIPTION
**Pull request by sfernandez11**
## Summary
* Added TS configuration to bootstrap generated project.
* Refactor on some files.
* Replaced old and unused files and tasks.
* Updated README
* Updated eslint and prettier configs (add .prettier config file)
* Fixed some linter problems
* Disabled eslint on `.ejs` files for now
* Updated `bitrise.yml`

## TODO
* Migration of all application files to TSX/TS.
* Create all necessary interfaces.

## Screenshots
Captura de Pantalla 2020-04-06 a la(s) 15 33 44

## Trello Card
https://trello.com/c/qpbQ0hpf/148-agregar-typescript-al-bootstrap

### CODE REVIEW

File: generators/app/templates/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+declare module '*.png';
+declare module '*.jpg';
+declare module '*.json';
+declare module 'reactotron-apisauce';
+declare module 'recompose';
+declare module 'redux-recompose';
+declare module 'redux-persist-seamless-immutable';
```
Comment (felire) [inline generators/app/templates/index.d.ts:7]: this is optional, it comes with `redux-persist` feature

File: generators/app/templates/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+declare module 'react-native-push-notification';
+declare module '@react-native-firebase/messaging';
```
Comment (felire) [inline generators/app/templates/index.d.ts:9]: I think those are optional too

File: generators/app/templates/index.d.ts (hunk 1 above)
Comment (sfernandez11) [reply generators/app/templates/index.d.ts]: Done

File: generators/app/templates/index.d.ts (hunk 2 above)
Comment (sfernandez11) [reply generators/app/templates/index.d.ts]: Done

File: generators/app/templates/index.d.ejs [hunk 3]
```diff
@@ -1,9 +1,15 @@
-declare module '*.png';
 declare module '*.jpg';
 declare module '*.json';
-declare module 'reactotron-apisauce';
```
Comment (ignaciosantise) [inline generators/app/templates/index.d.ejs]: bye bye reactotron?

Comment (sfernandez11) [reply generators/app/templates/index.d.ejs]: My bad, nice catch hahaha

**Pull request merged**